### ISSUE TITLE
Switch from GPS plugin dependency to framework

### ISSUE THREAD

**Issue by john-gu** [danwilson/google-analytics-plugin#120]
The &lt;dependency id="com.google.playservices@19.0.0" /&gt; is deprecated:
https://github.com/MobileChromeApps/google-play-services

With Cordova Android 4.0.0+, this can be replaced by a framework (also requires installing Google Repository from Android SDK):
&lt;framework src="com.google.android.gms:play-services:+" /&gt;

Using a framework should also resolve lib conflicts with other plugins using GPS:
https://github.com/danwilson/google-analytics-plugin/issues/117
https://github.com/danwilson/google-analytics-plugin/issues/96
https://github.com/danwilson/google-analytics-plugin/issues/76

EDIT: also need to include version reference that was previously included by the dependency plugin:
https://github.com/MobileChromeApps/google-play-services/blob/master/plugin.xml#L28

**Comment by guidokessels** [comment]
+1

**Comment by Daz66** [comment]
+1

**Comment by quanghoc** [comment]
Hi

Do you have the steps to do this? I modify the xml file and removed the google playservices plugin. My build still failed. I guess I missed "something" in this comment?

`also requires installing Google Repository from Android SDK`

**Comment by john-gu** [comment]
What's the error message you're getting?

**Comment by quanghoc** [comment]
Not sure where to copy the error but here:

```
UNEXPECTED TOP-LEVEL EXCEPTION:
com.android.dex.DexException: Multiple dex files define Lcom/google/ads/AdRequest$ErrorCode;
	at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
	at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
	at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
	at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
	at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
	at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:454)
	at com.android.dx.command.dexer.Main.runMonoDex(Main.java:303)
	at com.android.dx.command.dexer.Main.run(Main.java:246)
	at com.android.dx.command.dexer.Main.main(Main.java:215)
	at com.android.dx.command.Main.main(Main.java:106)

 FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dexDebug'.
> com.android.ide.common.internal.LoggedErrorException: Failed to run command:
  	/Users/phanh1/Library/Android/sdk/build-tools/22.0.1/dx --dex --no-optimize --output /Users/phanh1/google-analytics-plugin-test/platforms/android/build/intermediates/dex/debug --input-list=/Users/phanh1/google-analytics-plugin-test/platforms/android/build/intermediates/tmp/dex/debug/inputList.txt
  Error Code:
  	2
  Output:
  	
  	UNEXPECTED TOP-LEVEL EXCEPTION:
  	com.android.dex.DexException: Multiple dex files define Lcom/google/ads/AdRequest$ErrorCode;
  		at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:596)
  		at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:554)
  		at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:535)
  		at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
  		at com.android.dx.merge.DexMerger.merge(DexMerger.java:189)
  		at com.android.dx.command.dexer.Main.mergeLibraryDexBuffers(Main.java:454)
  		at com.android.dx.command.dexer.Main.runMonoDex(Main.java:303)
  		at com.android.dx.command.dexer.Main.run(Main.java:246)
  		at com.android.dx.command.dexer.Main.main(Main.java:215)
  		at com.android.dx.command.Main.main(Main.java:106)
  	


* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 2 mins 53.982 secs

/Users/phanh1/google-analytics-plugin-test/platforms/android/cordova/node_modules/q/q.js:126
                    throw e;
                          ^
Error code 1 for command: /Users/phanh1/google-analytics-plugin-test/platforms/android/gradlew with args: cdvBuildDebug,-b,/Users/phanh1/google-analytics-plugin-test/platforms/android/build.gradle,-Dorg.gradle.daemon=true
ERROR building one of the platforms: Error: /Users/phanh1/google-analytics-plugin-test/platforms/android/cordova/build: Command failed with exit code 1
You may not have the required environment or OS to build this project
Error: /Users/phanh1/google-analytics-plugin-test/platforms/android/cordova/build: Command failed with exit code 1
    at ChildProcess.whenDone (/usr/local/lib/node_modules/cordova/node_modules/cordova-lib/src/cordova/superspawn.js:134:23)
    at ChildProcess.emit (events.js:110:17)
    at maybeClose (child_process.js:1015:16)
    at Process.ChildProcess._handle.onexit (child_process.js:1087:5)

```

**Comment by john-gu** [comment]
If you type 'cordova plugin', do you see the old 'com.google.playservices' plugin still in your list of plugins?

You need to rm that plugin, as well as remove it from the 'dependent_plugins' entry in plugins/android.json. 'cordova plugin rm' does not remove dependencies.

**Comment by quanghoc** [comment]
I completed delete everything to start from scratch using this folk https://github.com/t-nonque/google-analytics-plugin and the build works. However, I got no data into GA at yet. But that is a different problem. Thanks.

**Comment by john-gu** [comment]
I think with the removal of the dependency plugin, you will also need to add this line to the plugin.xml under parent '/manifest/application':
https://github.com/MobileChromeApps/google-play-services/blob/master/plugin.xml#L28

as described in step 4 here:
https://developers.google.com/android/guides/setup

**Comment by danwilson** [comment]
Closed with #118
